<launch>
    <arg name="port" default="/dev/ttyUSB1"/>
    <arg name="baud_rate" default="11520"/>
    <arg name="odom_frame" default="odom"/>
    <arg name="base_frame" default="base_link"/>
    <arg name="control_rate" default="30"/>
    <arg name="odom_rate" default="30"/>
    <arg name="sensor_rate" default="5"/>
    <arg name="use_description" default="true"/>
    <arg name="use_tf" default="false"/>
    <arg name="use_control" default="true"/>
    <arg name="use_rviz" default="false"/>

    <param name="robot_description"
        textfile="$(find neurobot_description)/urdf/neurobot_v2.urdf" />

    <node name="robot_state_publisher" pkg="robot_state_publisher"
        type="robot_state_publisher" />

    <node pkg="neurobot_bringup" type="neurobot_bringup_node" name="neurobot_bringup" output="screen">
        <param name="port_name" type="str" value="$(arg port)"/>
        <param name="baud_rate" type="str" value="$(arg baud_rate)"/>
        <param name="odom_frame" type="str" value="$(arg odom_frame)"/>
        <param name="base_frame" type="str" value="$(arg base_frame)"/>
        <param name="control_rate" type="int" value="$(arg control_rate)"/>
        <param name="odom_rate" type="int" value="$(arg odom_rate)"/>
        <param name="sensor_rate" type="int" value="$(arg sensor_rate)"/>
        <param name="use_tf" type="bool" value="$(arg use_tf)"/>
        <param name="use_control" type="bool" value="$(arg use_control)"/>
    </node>

    <group if="$(arg use_rviz)">
        <node pkg="rviz" type="rviz" name="rviz" args="-d $(find neurobot_bringup)/rviz/neurobot.rviz" />
    </group>

</launch>